pages/index.tsx

```diff
@@ -0,0 +1,584 @@
+                  out: {
+                    data: [
+                      {
+                        type_id: containTypeLinkId,
+                        to: {
+                          data: {
+                            type_id: mimetypeTypeLinkId,
+                            string: { data: { value: sound.record["mimeType"] } },
+                          }
+                        }
+                      },
+                      {
+                        type_id: containTypeLinkId,
+                        to: {
+                          data: {
+                            type_id: formatTypeLinkId,
+                            string: { data: { value: sound.record["mimeType"] === "audio/webm;codecs=opus" ? "webm" : "aac" } },
+                          }
+                        }
+                      }
+                    ]
+                  }
+                }
+              }
+            },
+            {
+              type_id: containTypeLinkId,
+              to: {
+                data: {
+                  type_id: durationTypeLinkId,
+                  number: { data: { value: sound.record["msDuration"] } },
+                }
+              }
+            },
+            {
+              type_id: containTypeLinkId,
+              to: {
+                data: {
+                  type_id: startTimeTypeLinkId,
+                  string: { data: { value: sound.startTime } },
+                }
+              }
+            },
+            {
+              type_id: containTypeLinkId,
+              to: {
+                data: {
+                  type_id: endTimeTypeLinkId,
+                  string: { data: { value: sound.endTime } },
+                }
+              }
+            }]
+        }
+      })));
+      console.log("flakeed1");
+      console.log("soundLinkId", soundLinkId)
+
+      const { data: [isGcloudAuthKeyLinkId] } = await deep.select({
+        type_id: gcloudAuthKeyTypeLink,
+        in: {
+          type_id: containTypeLinkId,
+          from_id: deep.linkId,
+        },
+      });
+
+      console.log("googleauth", googleAuth);
+      if (!isGcloudAuthKeyLinkId) {
+        const parsedGoogleAuth = JSON.parse(googleAuth);
+        await deep.insert({
+          type_id: gcloudAuthKeyTypeLink,
+          object: { data: { value: parsedGoogleAuth } },
+          in: {
+            data: [
+              {
+                type_id: containTypeLinkId,
+                from_id: deep.linkId,
+              }
+            ]
+          }
+        })
+      }
+      console.log("flakeed3");
+
+      const { data: [{ id: transcribeTextLinkId }] } = await deep.insert({
+        type_id: transcribeTypeLinkId,
+        from_id: deep.linkId,
+        to_id: soundLinkId,
+        in: {
+          data: {
+            type_id: containTypeLinkId,
+            from_id: deep.linkId,
+          }
+        }
+      });
+      console.log("transcribeTextLinkId", transcribeTextLinkId)
+
+      await delay(8000);
```

Comment:
Do not wait fixed amount of time. Do a loop where you select and wait one second if data is not found. Loop must be executed n (for example 10) times
Smth like this:
```ts
let transcribedTextLinkId;
for(const i = 0; i < 10, i++) {
if(transcribedTextLinkId) break;
transcribedTextLinkId = /* select */
}
if(!transcribedTextLinkId) throw new Error(/*message*/)
```